diff -urN gnome-keyring-0.3.1.orig/configure.in gnome-keyring-0.3.1/configure.in
--- gnome-keyring-0.3.1.orig/configure.in	2004-07-05 10:55:37.402735176 +0200
+++ gnome-keyring-0.3.1/configure.in	2004-07-05 10:55:51.227633472 +0200
@@ -33,7 +33,7 @@
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [The gettext domain name])
 AC_SUBST(GETTEXT_PACKAGE)
 
-ALL_LINGUAS="ar az bg ca cs cy da de el en_CA en_GB es eu fi fr hr hu id it is ja ko lt mn ms nl no pa pl pt pt_BR ru sk sq sr sr@Latn sv th tr uk zh_CN zh_TW"
+ALL_LINGUAS="ar az bg ca cs cy da de el en_CA en_GB es eu fi fr hr hu id it is ja ko lt mn ms nb nl pa pl pt pt_BR ru sk sq sr sr@Latn sv th tr uk zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 dnl check for socklen_t
