diff -urN aa/gnome-keyring-0.1.90/configure.in gnome-keyring-0.1.90/configure.in
--- aa/gnome-keyring-0.1.90/configure.in	2004-03-08 12:25:30.000000000 +0100
+++ gnome-keyring-0.1.90/configure.in	2004-03-08 13:20:39.317669576 +0100
@@ -33,7 +33,7 @@
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [The gettext domain name])
 AC_SUBST(GETTEXT_PACKAGE)
 
-ALL_LINGUAS="ar az ca cs cy da de el en_CA en_GB es fi fr hr hu it is ja ko lt mn ms nl no pa pl pt pt_BR ru sk sq sr sr@Latn sv th tr uk zh_CN zh_TW"
+ALL_LINGUAS="ar az ca cs cy da de el en_CA en_GB es fi fr hr hu it is ja ko lt mn ms nb nl pa pl pt pt_BR ru sk sq sr sr@Latn sv th tr uk zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 dnl check for socklen_t
